### PR TITLE
Enable SMS OTP authenticator in Suborgs

### PR DESCRIPTION
**Pull request by PasinduYeshan**
### Purpose
$subject

### Related Issues
- https://github.com/wso2/product-is/issues/18247

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/extensions/configs/authenticator.tsx [hunk 1]
```diff
@@ -21,11 +21,12 @@ import { AuthenticatorConfig } from "./models";
 import { ChoreoButton } from "../components/shared/button/choreo-navigation-button";
 
 export const authenticatorConfig : AuthenticatorConfig = {
+    disableSMSOTPInSubOrgs: false,
```
Comment (pavinduLakshan) [inline apps/console/src/extensions/configs/authenticator.tsx:24]: Let's send the cloud support PR as well.

File: apps/console/src/features/applications/components/settings/sign-on-methods/sign-in-method-landing.tsx [hunk 2]
```diff
@@ -17,6 +17,7 @@
  */
 import { IdentifiableComponentInterface, SBACInterface } from "@wso2is/core/models";
 import { Heading, InfoCard, useMediaContext } from "@wso2is/react-components";
+import { authenticatorConfig } from "apps/console/src/extensions";
```
Comment (pavinduLakshan) [inline apps/console/src/features/applications/components/settings/sign-on-methods/sign-in-method-landing.tsx:20]: Let's import from the relative paths

File: apps/console/src/features/applications/components/settings/sign-on-methods/sign-in-method-landing.tsx [hunk 3]
```diff
@@ -196,7 +197,8 @@ export const SignInMethodLanding: FunctionComponent<SignInMethodLandingPropsInte
                             ) }
                             {
                                 (!hiddenOptions.includes(LoginFlowTypes.SECOND_FACTOR_SMS_OTP) &&
-                                    organizationType !== OrganizationType.SUBORGANIZATION) && (
+                                    !(organizationType === OrganizationType.SUBORGANIZATION &&
+                                    authenticatorConfig.disableSMSOTPInSubOrgs)) && (
```
Comment (pavinduLakshan) [inline apps/console/src/features/applications/components/settings/sign-on-methods/sign-in-method-landing.tsx:201]:
```suggestion
                                    authenticatorConfig?.disableSMSOTPInSubOrgs)) && (
```

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/add-authenticator-modal.tsx [hunk 4]
```diff
@@ -245,7 +245,8 @@ export const AddAuthenticatorModal: FunctionComponent<AddAuthenticatorModalProps
         ];
 
         // Remove SMS OTP authenticator from the list at the sub org level.
-        _filteredAuthenticators = (orgType === OrganizationType.SUBORGANIZATION)
+        _filteredAuthenticators = (orgType === OrganizationType.SUBORGANIZATION &&
+            authenticatorConfig.disableSMSOTPInSubOrgs)
```
Comment (pavinduLakshan) [inline apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/add-authenticator-modal.tsx:249]:
```suggestion
            authenticatorConfig?.disableSMSOTPInSubOrgs)
```

File: apps/console/src/features/connections/pages/connections.tsx [hunk 5]
```diff
@@ -28,19 +28,20 @@ import {
...
 } from "react";
 import { useTranslation } from "react-i18next";
 import { useSelector } from "react-redux";
 import { Icon } from "semantic-ui-react";
-import { 
+import {
     AuthenticatorExtensionsConfigInterface,
     identityProviderConfig
 } from "../../../extensions/configs";
+import { authenticatorConfig as extensionsAuthenticatorConfig } from "../../../extensions/configs/authenticator";
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/pages/connections.tsx:44]:
```suggestion
import { authenticatorConfig as extendedAuthenticatorConfig } from "../../../extensions/configs/authenticator";
```

File: apps/console/src/features/connections/pages/connections.tsx [hunk 6]
```diff
@@ -222,7 +223,8 @@ const ConnectionsPage: FC<ConnectionsPropsInterface> = (props: ConnectionsPropsI
 
             // Hide the SMS OTP authenticator for sub organizations.
             if (authenticator.id === AuthenticatorManagementConstants.SMS_OTP_AUTHENTICATOR_ID &&
-                organizationType === OrganizationType.SUBORGANIZATION) {
+                organizationType === OrganizationType.SUBORGANIZATION &&
+                extensionsAuthenticatorConfig.disableSMSOTPInSubOrgs) {
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/pages/connections.tsx:227]: Plz check and update the other places as well

```suggestion
                extensionsAuthenticatorConfig?.disableSMSOTPInSubOrgs) {
```

File: apps/console/src/features/applications/components/settings/sign-on-methods/sign-in-method-landing.tsx (hunk 2 above)
Comment (PasinduYeshan) [reply apps/console/src/features/applications/components/settings/sign-on-methods/sign-in-method-landing.tsx]: Changed

File: apps/console/src/extensions/configs/authenticator.tsx (hunk 1 above)
Comment (PasinduYeshan) [reply apps/console/src/extensions/configs/authenticator.tsx]: Done

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/add-authenticator-modal.tsx (hunk 4 above)
Comment (PasinduYeshan) [reply apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/add-authenticator-modal.tsx]: Changed

File: apps/console/src/features/connections/pages/connections.tsx (hunk 6 above)
Comment (PasinduYeshan) [reply apps/console/src/features/connections/pages/connections.tsx]: Changed

**Review by pavinduLakshan — DISMISSED** [review]

File: apps/console/src/features/connections/pages/connections.tsx (hunk 5 above)
Comment (PasinduYeshan) [reply apps/console/src/features/connections/pages/connections.tsx]: Changed

**Review by pavinduLakshan — APPROVED** [review]

**Pull request merged**